fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingPageState.kt

```diff
@@ -13,19 +13,19 @@ import org.mozilla.fenix.compose.LinkTextState
  * @property imageRes [DrawableRes] displayed on the page.
  * @property title [String] title of the page.
  * @property description [String] description of the page.
- * @property linkTextState [LinkTextState] part of description text with a link.
  * @property primaryButton [Action] action for the primary button.
  * @property secondaryButton [Action] action for the secondary button.
  * @property onRecordImpressionEvent Callback for recording impression event.
+ * @property privacyCaption privacy caption to show and allow user to view on privacy policy.
  */
 data class OnboardingPageState(
     @DrawableRes val imageRes: Int,
     val title: String,
     val description: String,
-    val linkTextState: LinkTextState? = null,
     val primaryButton: Action,
     val secondaryButton: Action? = null,
     val onRecordImpressionEvent: () -> Unit = {},
+    val privacyCaption: Caption?,
```

Comment:
We could set null as a default so we don't have to set it in `UpgradeOnboarding` and `NotificationDialogScreen`. Also, maybe we can use the same param position as now deleted `linkTextState` so lamda is the last param